src/src/radio.c

```diff
@@ -3053,58 +3149,66 @@ static bool validatePacketTimestamp(RAIL_RxPacketDetails_t *pPacketDetails, uint
     // from 0x00000001FFFFFFFF to 0x0000000200000000 (leave the higher 32 bits as 0)
-    if ((railUsTimeNow & 0xFFFFFFFF) <= pPacketDetails->timeReceived.packetTime)
+    if ((railUsTimeNow & 0xFFFFFFFF) <= timestamp)
     {
         railUsTimerWraps--;
     }
 
-    if (pPacketDetails->isAck)
+    if (pPacketDetails != NULL && pPacketDetails->isAck)
     {
-        sReceiveAckFrame.mInfo.mRxInfo.mRssi = pPacketDetails->rssi;
-        sReceiveAckFrame.mInfo.mRxInfo.mLqi  = pPacketDetails->lqi;
-        sReceiveAckFrame.mInfo.mRxInfo.mTimestamp =
-            pPacketDetails->timeReceived.packetTime + ((uint64_t)railUsTimerWraps << 32);
-// TODO: Maybe remove #if ?
+        sReceiveAckFrame.mInfo.mRxInfo.mRssi      = pPacketDetails->rssi;
+        sReceiveAckFrame.mInfo.mRxInfo.mLqi       = pPacketDetails->lqi;
+        sReceiveAckFrame.mInfo.mRxInfo.mTimestamp = timestamp + ((uint64_t)railUsTimerWraps << 32);
 #if OPENTHREAD_RADIO && OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE == 1
         sReceiveAckFrame.mIid = iid;
 #endif
     }
     else
     {
-        sReceiveFrame[sReceiveBufferInUse].mInfo.mRxInfo.mRssi = pPacketDetails->rssi;
-        sReceiveFrame[sReceiveBufferInUse].mInfo.mRxInfo.mLqi  = pPacketDetails->lqi;
-        sReceiveFrame[sReceiveBufferInUse].mInfo.mRxInfo.mTimestamp =
-            pPacketDetails->timeReceived.packetTime + ((uint64_t)railUsTimerWraps << 32);
+        if (pPacketDetails != NULL)
+        {
+            sReceiveFrame.mInfo.mRxInfo.mRssi = pPacketDetails->rssi;
+            sLastRssi                         = pPacketDetails->rssi;
+
+            sReceiveFrame.mInfo.mRxInfo.mLqi = pPacketDetails->lqi;
+            sLastLqi                         = pPacketDetails->lqi;
+        }
+        else
+        {
+            sReceiveFrame.mInfo.mRxInfo.mRssi = sReceivePacket[aReceiveBufferInUse].packetInfo.rssi;
+            sLastRssi                         = sReceivePacket[aReceiveBufferInUse].packetInfo.rssi;
+
+            sReceiveFrame.mInfo.mRxInfo.mLqi = sReceivePacket[aReceiveBufferInUse].packetInfo.lqi;
+            sLastLqi                         = sReceivePacket[aReceiveBufferInUse].packetInfo.rssi;
+        }
+
+        sReceiveFrame.mInfo.mRxInfo.mTimestamp = timestamp + ((uint64_t)railUsTimerWraps << 32);
         // Set this flag only when the packet is really acknowledged with a secured enhanced ACK.
-        sReceiveFrame[sReceiveBufferInUse].mInfo.mRxInfo.mAckedWithSecEnhAck = securedOutgoingEnhancedAck;
+        sReceiveFrame.mInfo.mRxInfo.mAckedWithSecEnhAck = securedOutgoingEnhancedAck;
         // Set this flag only when the packet is really acknowledged with frame pending set.
-        sReceiveFrame[sReceiveBufferInUse].mInfo.mRxInfo.mAckedWithFramePending = framePendingSetInOutgoingAck;
+        sReceiveFrame.mInfo.mRxInfo.mAckedWithFramePending = framePendingSetInOutgoingAck;
 #if OPENTHREAD_RADIO && OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE == 1
```

Comment:
GSDK version does not have #if wrapper around   `sReceiveFrame.mIid = iid; `

Reply:
You can ignore this comment now. I understand why there is a required difference.